src/IconBuilder.php

```diff
@@ -0,0 +1,104 @@
+<?php declare(strict_types=1);
+
+namespace Bugo\FontAwesomeHelper;
+
+use InvalidArgumentException;
+use Nette\Utils\Html;
+
+use function array_merge;
+use function implode;
+use function str_starts_with;
+use function in_array;
+
+class IconBuilder
+{
+    private array $allowedSizes = [
```

Comment:
Why was a Enums\Size not introduced?

Enums can also be used for verifying incoming data.

Say for example your ->size(string $size) method could be typed for (Size|string $size)

if(! is_string($size)) {
    return $this->addClass('fa-', $size->value); // we know we have a valid value
}
$size = Size::tryFrom($size) ?? Size::Default->value; // Where Default can be an Enum constant alias pointing to whichever size you want to be the default.

public final const Default = '1x';

case 1x = self::Default;

At which time, instead of throwing that exception you could provide real time user feedback that they have entered an incorrect value and that the default will be used along with providing a list of possible values, provided via the Enum without the need for a trip to a db.

Dont get me wrong. The current code works :) The point of Enums is to always work with known types and values :) I have not benchmarked it but I would guess that tryFrom is faster than in_array. I should probably bench that come to think of it.